recipes/msys2/all/conanfile.py

```diff
@@ -169,18 +178,23 @@ def package(self):
     def package_info(self):
         self.cpp_info.libdirs = []
         self.cpp_info.includedirs = []
+        self.cpp_info.resdirs = []
 
         msys_root = self._msys_dir
         msys_bin = os.path.join(msys_root, "usr", "bin")
+        self.cpp_info.bindirs.append(msys_bin)
 
-        self.output.info("Creating MSYS_ROOT env var : %s" % msys_root)
-        self.env_info.MSYS_ROOT = msys_root
+        self.output.info(f"Creating MSYS_ROOT env var : {msys_root}")
+        self.buildenv_info.define_path("MSYS_ROOT", msys_root)
 
-        self.output.info("Creating MSYS_BIN env var : %s" % msys_bin)
-        self.env_info.MSYS_BIN = msys_bin
+        self.output.info(f"Creating MSYS_BIN env var : {msys_bin}")
+        self.buildenv_info.define_path("MSYS_BIN", msys_bin)
+
+        self.conf_info.define("tools.microsoft.bash:subsystem", "msys2")
```

Comment:
Nice!!!

Reply:
I'm also experimenting with changing this recipe and using the `buildenv_info`, but I noticed that I needed to fix the subsystem.py for that to work correctly in consuming projects. See PR conan-io/conan#11981 and conan-io/conan#11980

I also still needed to keep on using the old `env_info` otherwise to much of my dependencies such as `libffi` needed to be overhauled. By defining them both I only needed to overhaul automake and autoconf. See: https://github.com/conan-io/conan-center-index/compare/master...Ultimaker:conan-center-index:modernize_autoconf for a work in progress

Reply:
> I also still needed to keep on using the old `env_info` otherwise to much of my dependencies such as `libffi` needed to be overhauled. By defining them both I only needed to overhaul automake and autoconf. See: [master...Ultimaker:conan-center-index:modernize_autoconf](https://github.com/conan-io/conan-center-index/compare/master...Ultimaker:conan-center-index:modernize_autoconf) for a work in progress

related to https://github.com/conan-io/conan/issues/11974#issuecomment-1229553132?

Reply:
The fact is that `env_info` must be kept because conan v1 recipes don't use at all `buildenv_info` of tool_requires, and even some conan v2 recipes (it depends of user configuration, see https://github.com/conan-io/conan/issues/11974#issuecomment-1229553132 again).

Reply:
Thx, will keep that in mind. You're updating those recipes at an amazing speed. I think that when I'm finally finished with my struggle getting CPython and Qt to compile on Windows, my change aren't needed anymore 😉.